src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Common/Builder.cs

```diff
@@ -507,12 +512,8 @@ protected bool BuildInternal (string projectOrSolution, string target, string []
 			}
 			if (!result && ThrowOnBuildFailure) {
 				string message = "Build failure: " + Path.GetFileName (projectOrSolution) + (BuildLogFile != null && File.Exists (buildLogFullPath) ? "Build log recorded at " + buildLogFullPath : null);
-				//NOTE: enormous logs will lock up IDE's UI
-				if (IsRunningInIDE) {
-					throw new FailedBuildException (message);
-				} else {
-					throw new FailedBuildException (message, null, File.ReadAllText (buildLogFullPath));
```

Comment:
I assume this will now be handled by the AddTestAttachement call?

Reply:
Assuming that these logs are output in the folder we are looking for then yes, though maybe we should add a similar call here as well if `TestContext` is not null.